cypress/support/user.ts

```diff
@@ -39,19 +55,87 @@ const testCredentialStore = {
   },
 };
 
+function changeActiveRole(selectedRoleId: number) {
+  const token = window.localStorage.getItem('token');
+
+  if (!token) {
+    throw new Error('No logged in user');
+  }
+
+  const api = getE2EApi();
+  const request = api.selectRole({ selectedRoleId, token }).then((resp) => {
+    if (!resp.selectRole.token) {
+      return;
+    }
+
+    const { currentRole, user, exp } = jwtDecode(
+      resp.selectRole.token
+    ) as DecodedTokenData;
+
+    window.localStorage.setItem('token', resp.selectRole.token);
+    window.localStorage.setItem(
+      'currentRole',
+      currentRole.shortCode.toUpperCase()
+    );
+    window.localStorage.setItem('expToken', `${exp}`);
+    window.localStorage.setItem('user', JSON.stringify(user));
+  });
+
+  cy.wrap(request);
+}
+
+const externalTokenLogin = (
+  roleOrCredentials: 'user' | 'officer' | 'user2' | 'placeholderUser'
+): Cypress.Chainable<LoginMutation | ExternalTokenLoginMutation> => {
+  const credentials = testCredentialStoreStfc[roleOrCredentials];
+
+  const api = getE2EApi();
+  const request = api.externalTokenLogin(credentials).then((resp) => {
+    if (!resp.externalTokenLogin.token) {
+      return resp;
+    }
+
+    const { user, exp } = jwtDecode(
+      resp.externalTokenLogin.token
+    ) as DecodedTokenData;
+
+    window.localStorage.setItem('token', resp.externalTokenLogin.token);
+    window.localStorage.setItem(
+      'currentRole',
+      roleOrCredentials === 'user' ? 'USER' : 'USER_OFFICER'
```

Comment:
It would be better to default to the `USER` role if the credentials weren't specifically for a user officer, otherwise all credentials aside from `user` will get user officer permissions:

```suggestion
      roleOrCredentials === 'officer' ? 'USER_OFFICER' : 'USER'
```